api/maistra/v1alpha1/session_types.go

```diff
@@ -88,6 +88,50 @@ type SessionStatus struct {
 	RouteExpression string   `json:"_routeExp,omitempty"`   //nolint:tagliatelle //reason intentionally prefixed with _ to distinguish as UI/CLI field.
 	Strategies      []string `json:"_strategies,omitempty"` //nolint:tagliatelle //reason intentionally prefixed with _ to distinguish as UI/CLI field.
 	RefNames        []string `json:"_refNames,omitempty"`   //nolint:tagliatelle //reason intentionally prefixed with _ to distinguish as UI/CLI field.
+
+	Readiness StatusReadiness `json:"readiness,omitempty"`
+}
+
+type StatusReadiness struct {
+	// Status of resources deployed/modifed by this Session resource
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Resources", xDescriptors="urn:alm:descriptor:com.tectonic.ui:podStatuses"
+	Components StatusComponents `json:"components,omitempty"`
+}
+
+type StatusComponents struct {
+	Pending []string `json:"pending,omitempty"`
+	Ready   []string `json:"ready,omitempty"`
+	UnReady []string `json:"unready,omitempty"`
+}
+
+func (s *StatusComponents) SetPending(comp string) {
+	s.Ready = s.removeFrom(s.Ready, comp)
+	s.UnReady = s.removeFrom(s.UnReady, comp)
+	s.Pending = append(s.Pending, comp)
+}
+
+func (s *StatusComponents) SetReady(comp string) {
+	s.Pending = s.removeFrom(s.Pending, comp)
+	s.UnReady = s.removeFrom(s.UnReady, comp)
+	s.Ready = append(s.Ready, comp)
+}
```

Comment:
It looks fine and stuff, but maybe some unit tests with boundary conditions would help from regressions in the future?

Reply:
good point

Reply:
Added unit and int level tests https://github.com/maistra/istio-workspace/pull/875/commits/ccdb745a662402b43b249e07ff6a4d37100ca6f7